plugin/evm/vm_test.go

```diff
@@ -2052,3 +2054,77 @@ func TestBlockGasValidation(t *testing.T) {
 		})
 	}
 }
+
+func TestMinDelayExcessInHeader(t *testing.T) {
+	tests := []struct {
+		name                   string
+		fork                   upgradetest.Fork
+		desiredMinDelay        *uint64
+		expectedMinDelayExcess *acp226.DelayExcess
+	}{
+		{
+			name:                   "pre_granite_no_min_delay_excess",
+			fork:                   upgradetest.Fortuna,
+			desiredMinDelay:        nil,
+			expectedMinDelayExcess: nil,
+		},
+		{
+			name:                   "pre_granite_min_delay_excess",
+			fork:                   upgradetest.Fortuna,
+			desiredMinDelay:        utils.NewUint64(1000),
+			expectedMinDelayExcess: nil,
+		},
+		{
+			name:                   "granite_first_block_initial_delay_excess",
+			fork:                   upgradetest.Granite,
+			desiredMinDelay:        nil,
+			expectedMinDelayExcess: customheadertest.NewDelayExcessPtr(acp226.InitialDelayExcess),
+		},
+		{
+			name:                   "granite_with_excessive_desired_min_delay_excess",
+			fork:                   upgradetest.Granite,
+			desiredMinDelay:        utils.NewUint64(4000),
+			expectedMinDelayExcess: customheadertest.NewDelayExcessPtr(acp226.InitialDelayExcess + acp226.MaxDelayExcessDiff),
+		},
+		{
+			name:                   "granite_with_zero_desired_min_delay_excess",
+			fork:                   upgradetest.Granite,
+			desiredMinDelay:        utils.NewUint64(0),
+			expectedMinDelayExcess: customheadertest.NewDelayExcessPtr(acp226.InitialDelayExcess - acp226.MaxDelayExcessDiff),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			require := require.New(t)
+			ctx := context.Background()
+			var configJSON string
+			if test.desiredMinDelay != nil {
+				// convert excess to delay
+				configJSON = fmt.Sprintf(`{"min-delay-target": %d}`, *test.desiredMinDelay)
+			}
+			vm := newDefaultTestVM()
+			vmtest.SetupTestVM(t, vm, vmtest.TestVMConfig{
+				Fork:       &test.fork,
+				ConfigJSON: configJSON,
+			})
+
+			defer func() {
+				require.NoError(vm.Shutdown(ctx))
+			}()
+
+			// Build a block
+			signedTx := newSignedLegacyTx(t, vm.chainConfig, vmtest.TestKeys[0].ToECDSA(), 0, &vmtest.TestEthAddrs[1], big.NewInt(1), 21000, vmtest.InitialBaseFee, nil)
+			blk, err := vmtest.IssueTxsAndBuild([]*types.Transaction{signedTx}, vm)
+			require.NoError(err)
+
+			// Check the min delay excess in the header
+			ethBlock := blk.(*chain.BlockWrapper).Block.(*wrappedBlock).ethBlock
+			headerExtra := customtypes.GetHeaderExtra(ethBlock.Header())
+
+			require.Equal(test.expectedMinDelayExcess, headerExtra.MinDelayExcess, "expected %s, got %s",
+				utils.SafeDerefUint64String(utils.Uint64PtrFrom(test.expectedMinDelayExcess)),
+				utils.SafeDerefUint64String(utils.Uint64PtrFrom(headerExtra.MinDelayExcess)))
+		})
```

Comment:
imo, I feel like just:
```suggestion
			require.Equal(test.expectedMinDelayExcess, headerExtra.MinDelayExcess, "expected %d, got %d",
				test.expectedMinDelayExcess,
				headerExtra.MinDelayExcess)
		})
```
is enough to allow the output to be readable (without introducing the `utils.SafeDerefUint64String` helper)... But if we really want to keep it I feel like we could make it a test helper.

Reply:
it's indeed annoying to debug...